### ISSUE TITLE
With only one class to detect, 500 iterations gives me an avg loss of 0.8. Do i still need to extend the training to 2000 iterations?

### ISSUE THREAD

**Issue by joelmatt** [AlexeyAB/darknet#4166]

**Comment by joelmatt** [comment]
using the map command:
`!../../.././darknet detector map data/obj.data yolo-obj.cfg backup/yolo-obj_last.weights -points 0`
and this is the output i get is
```
calculation mAP (mean average precision)...
320
 detections_count = 1317, unique_truth_count = 318  
class_id = 0, name = pothole, ap = 1.92%   	 (TP = 0, FP = 0) 

 for conf_thresh = 0.25, precision = -nan, recall = 0.00, F1-score = -nan 
 for conf_thresh = 0.25, TP = 0, FP = 0, FN = 318, average IoU = 0.00 % 

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
 mean average precision (mAP@0.50) = 0.019180, or 1.92 % 
Total Detection Time: 44.000000 Seconds
```
can you help me figure out how to evaluate these statements. 
is 1.92% the accuracy of the model ?

**Comment by bagminer** [comment]
To understand ap and map better, check this link - https://sanchom.wordpress.com/tag/average-precision/

If you observe, `class_id = 0, name = pothole, ap = 1.92%   	 (TP = 0, FP = 0) ` , there are 0 True Positive and False Positives. This means 0 'potholes' were detected out of 318 ( `for conf_thresh = 0.25, TP = 0, FP = 0, FN = 318, average IoU = 0.00 %`) . 
If you have a huge dataset, there is possibility that for that batch there might not be enough 'potholes' to detect. So train more. Also just to be sure, check whether the sanctity of the images, and also the logic of detecting that one class.

**Comment by joelmatt** [comment]
```

 calculation mAP (mean average precision)...
320
 detections_count = 490, unique_truth_count = 318  
class_id = 0, name = pothole, ap = 44.22%   	 (TP = 117, FP = 36) 

 for conf_thresh = 0.25, precision = 0.76, recall = 0.37, F1-score = 0.50 
 for conf_thresh = 0.25, TP = 117, FP = 36, FN = 201, average IoU = 54.80 % 

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
 mean average precision (mAP@0.50) = 0.442151, or 44.22 % 
Total Detection Time: 42.000000 Seconds
```
with close to 1500 images for the only one class to train, what else should be done to improve the the mAP scores.  I am close to 3200 iterations. Should i continue the training or stop and make changes somewhere

**Comment by bagminer** [comment]
First check for the training loss (avg). And then if the training loss is not decreasing much, then check for the highest mAP. 
In fact, to improve mAP (above 50-60%) you should validate if your ground truths are accurately and uniformly done for all the training image dataset. 
Refer - [https://github.com/AlexeyAB/darknet#when-should-i-stop-training];
[https://github.com/AlexeyAB/darknet#how-to-improve-object-detection]
